spinn_front_end_common/interface/interface_functions/locate_executable_start_type.py

```diff
@@ -51,17 +62,27 @@ def __call__(self, graph, placements, graph_mapper=None):
                         vertex, placements,
                         binary_start_types[placement_binary_start_type])
                 elif graph_mapper is not None:
-                    machine_verts = graph_mapper.get_machine_vertices(vertex)
+                    machine_verts = graph_mapper.get_machine_vertices(
+                        vertex)
                     for machine_vertex in machine_verts:
                         self._add_vertex_to_subset(
                             machine_vertex, placements,
-                            binary_start_types[placement_binary_start_type])
+                            binary_start_types[
+                                placement_binary_start_type])
+
+                # add to the binary to start type map
+                if binary_name is not None:
+                    binary_path = executable_finder.get_executable_path(
+                        binary_name)
+                    binary_to_start_type[binary_path] = (
+                        placement_binary_start_type)
 
-        # only got apps with no binary, such as external devices. return no app
+        # only got apps with no binary, such as external devices.
+        # return no app
         if not binary_start_types:
-            return [ExecutableType.NO_APPLICATION]
+            return [ExecutableType.NO_APPLICATION], {}
```

Comment:
Is this check needed?  Doesn't the line below just return an empty dict in this case?

Reply:
the check was already in master. so out of scope of the pr. the change is to make the return value correct in terms of expecting 2 elements and not 1. 

i assume that the check is needed. as without the explicit return, the return would be 2 empty dicts which is not the same as a explicit no application.

Reply:
OK, yes see it now